src/UI/CsobControl.php

```diff
@@ -146,6 +151,33 @@ public function pay()
 
 
 	/**
+	 * @param Csob\Payment $payment
+	 * @return Csob\Message\Response
+	 * @see https://github.com/csob/paymentgateway/wiki/eAPI-v1.6-EN#return-values-7
+	 */
+	private function oneclickPayment(Csob\Payment $payment)
+	{
+		// Note: We recommend to call payment/status not earlier than 2-3 seconds after payment/oneclick/start call.
+		// Should the payment status be still 2 (in progress), please poll the status every 5 seconds.
+		// The gateway depends performance of multiple systems beyond our control while authorising the transaction
+		// which can (in the worst case) take up to 60 seconds.
+
+		$initResponse = $this->client->paymentOneclickInit($payment);
+		$this->client->paymentOneclickStart($initResponse->getPayId());
+		sleep(3);
```

Comment:
Imho the polling should be separate method (accepting just `payId`), that will be not called here, but rather the user will call it manully, or move this easily to some (RPC) rabbit worker, etc